ui/src/components/dialog/__tests__/QDialog.cy.js

```diff
@@ -1,21 +1,131 @@
+import DialogWrapper from './DialogWrapper.vue'
+import { nextTick, ref } from 'vue'
+import { vModelAdapter } from '@quasar/quasar-app-extension-testing-e2e-cypress'
+
+function mountQDialogWrapper (options) {
+  return cy.mount(DialogWrapper, options)
+}
+
+function getHostElement () {
+  return cy.dataCy('dialog-card')
+}
+
+function closeDialogViaBackdrop () {
+  return cy.get('.q-dialog__backdrop').click({ force: true })
+}
+
+// TODO: this only works in `withinDialog` context,
+// as when using it outside it `.root()` will yeld "html" tag
+// and Cypress won't be able to find anything with `.closest('body')`
+// since it only searches upwards
+function closeDialogViaEscKey () {
+  // Official way to escape `within` context
+  // https://docs.cypress.io/api/commands/within#Temporarily-escape
+  return cy.root().closest('body').type('{esc}')
+}
+
+// TODO: we'll add this check to the AE helper to automatically test persistent dialogs
```

Comment:
This is okeish, but it doesn't account for seamless dialogs and dialogs without a backdrop
If we can find some other way to test for persistent dialogs being there, we can execute it automatically whenever a `withinDialog` command is called with `persistent: true`

Reply:
Checked the DOM again with these props and could not find any better way to identify persistent dialogs. We could probably add `cy.get('.q-dialog').should('not.have.class', 'q-dialog--seamless')` too but it is no guarantee that the persistent prop was set. So I left it as it is.

Reply:
Ok, we can't add this automatically into the Cypress AE helper then, but we can document this pattern and its limitations into the AE README, until we find a proper solution

Reply:
Documented these new helpers and their limitations here: https://github.com/quasarframework/quasar-testing/issues/355